Fix flag bypass for editing description

bmeneg opened the pull request:
While editing an MR or issue description if any flag different from -m and -F is used the editor is not opened. Add the --linebreak to this list of "accepted" flags to be used for editing the description.

Also, this MR bring other lateral fixes and improvements related to code organization, make sure to check each commit separately.

prarit reviewed (approved):
LGTM.  These are great cleanups.

bmeneg commented:
Of course, I forgot to update the test.

bmeneg commented:
@prarit the code behavior didn't change. I just updated the test code for including the description editing through file (`-F`) and fixed a variable declaration.
